packages/hooks-react/src/usePlaylist.ts

```diff
@@ -1,42 +1,78 @@
 
-export default function usePlaylist(playlistId?: string, params: GetPlaylistParams = {}, enabled: boolean = true, usePlaceholderData: boolean = true) {
+export const getPlaylistQueryOptions = ({
+  type,
+  contentId,
+  siteId,
+  enabled,
+  usePlaceholderData,
+  params = {},
+  queryClient,
+}: {
+  type: PlaylistMenuType;
+  contentId: string | undefined;
+  siteId: string;
+  enabled: boolean;
+  queryClient: QueryClient;
+  usePlaceholderData?: boolean;
+  params?: GetPlaylistParams;
+}) => {
   const apiService = getModule(ApiService);
-  const queryClient = useQueryClient();
-
-  const callback = async (playlistId?: string, params?: GetPlaylistParams) => {
-    const playlist = await apiService.getPlaylistById(playlistId, { ...params });
 
-    // This pre-caches all playlist items and makes navigating a lot faster.
-    playlist?.playlist?.forEach((playlistItem) => {
-      queryClient.setQueryData(['media', playlistItem.mediaid], playlistItem);
-    });
+  return {
+    enabled: !!contentId && enabled,
+    queryKey: ['playlist', type, contentId],
+    queryFn: async () => {
+      if (type === PLAYLIST_TYPE.playlist) {
+        const playlist = await apiService.getPlaylistById(contentId, params);
 
-    return playlist;
-  };
+        // This pre-caches all playlist items and makes navigating a lot faster.
+        playlist?.playlist?.forEach((playlistItem) => {
+          queryClient.setQueryData(['media', playlistItem.mediaid], playlistItem);
+        });
 
-  const queryKey = ['playlist', playlistId, params];
-  const isEnabled = !!playlistId && enabled;
+        return playlist;
+      } else if (type === PLAYLIST_TYPE.content_list) {
+        const contentList = await apiService.getContentList({ siteId, id: contentId });
 
-  return useQuery<Playlist | undefined, ApiError>(queryKey, () => callback(playlistId, params), {
-    enabled: isEnabled,
-    placeholderData: usePlaceholderData && isEnabled ? placeholderData : undefined,
-    refetchInterval: (data, _) => {
+        return contentList;
+      }
+    },
+    placeholderData: contentId && usePlaceholderData ? placeholderData : undefined,
+    refetchInterval: (data: Playlist | undefined) => {
       if (!data) return false;
 
       const autoRefetch = isTruthyCustomParamValue(data.refetch) || data.playlist.some(isScheduledOrLiveMedia);
 
       return autoRefetch ? 1000 * 30 : false;
     },
     retry: false,
-  });
+  };
+};
+
+export default function usePlaylist(
+  contentId?: string,
+  params: GetPlaylistParams = {},
+  enabled: boolean = true,
+  usePlaceholderData: boolean = true,
+  type: PlaylistMenuType = PLAYLIST_TYPE.playlist,
+) {
+  const queryClient = useQueryClient();
+  const siteId = useConfigStore((state) => state.config.siteId);
+
+  const queryOptions = getPlaylistQueryOptions({ type, contentId, siteId, params, queryClient, enabled, usePlaceholderData });
+
+  return useQuery<Playlist | undefined, ApiError>(queryOptions);
```

Comment:
You've gotta add usePlacehoderData and params inputs to usePlaylists, but then you avoid the getQueryOptions construct, which is a bit non-standard I think.

```suggestion
  return usePlaylists([{contentId, type}], enabled ? 1 : 0, usePlaceholderData, params)[0];
```

Reply:
I didn't want to do it because of the different use case we have here. usePlaylists is intended to work with a list of shelves on the Landing Page. So it also covers the cases of Favorites and Continue Watching, while usePlaylist only supports `content_list` and `playlist` types which makes the purpose a bit different.